src/layer/__tests__/outputResponses.ts

```diff
@@ -0,0 +1,73 @@
+import axios from 'axios';
+import MockAdapter from 'axios-mock-adapter';
+
+import { ApiType, setAuthToken, invalidateCaches, S2L2ALayer } from '../../index';
+
+import '../../../jest-setup';
+import { constructFixtureGetMapOutputResponses } from './fixtures.outputResponses';
+
+const mockNetwork = new MockAdapter(axios);
+
+const EXAMPLE_TOKEN = 'TOKEN111';
+
+beforeEach(async () => {
+  await invalidateCaches();
+});
+
+test('Error if outputResponseId is empty string in getMapParams', async () => {
+  const { layer, getMapParamsEmptyOutputResponseId } = constructFixtureGetMapOutputResponses();
+  setAuthToken(EXAMPLE_TOKEN);
+  try {
+    await layer.getMap(getMapParamsEmptyOutputResponseId, ApiType.PROCESSING);
+  } catch (e) {
+    expect(e.message).toBe('outputResponseId most not be empty');
+  }
+});
+
+test('Error if outputResponseId is used with WMS', async () => {
+  const { getMapParamsDefaultResponseId } = constructFixtureGetMapOutputResponses();
+  setAuthToken(EXAMPLE_TOKEN);
+  try {
+    const layer = new S2L2ALayer({ instanceId: 'INSTANCE_ID', layerId: 'LAYER_ID' });
+    await layer.getMap(getMapParamsDefaultResponseId, ApiType.WMS);
+  } catch (e) {
+    expect(e.message).toBe('outputResponseId is only available with Processing API');
+  }
+});
+
+test('Error if outputResponseId is empty and used with WMS', async () => {
+  const { getMapParamsEmptyOutputResponseId } = constructFixtureGetMapOutputResponses();
+  setAuthToken(EXAMPLE_TOKEN);
+  try {
+    const layer = new S2L2ALayer({ instanceId: 'INSTANCE_ID', layerId: 'LAYER_ID' });
+    await layer.getMap(getMapParamsEmptyOutputResponseId, ApiType.WMS);
+  } catch (e) {
+    expect(e.message).toBe('outputResponseId is only available with Processing API');
+  }
+});
+
+it('Success if outputResponseId is set correctly in getMapParams', async () => {
+  // arrayBuffer needs to be used, and removing this will cause getMap to fetch a blob, as window.Blob was created with jsdom
+  window.Blob = undefined;
+  const { layer, getMapParamsDefaultResponseId, mockedResponse } = constructFixtureGetMapOutputResponses();
+  setAuthToken(EXAMPLE_TOKEN);
+  mockNetwork.reset();
+  mockNetwork.onPost().replyOnce(200, mockedResponse);
+
+  await layer.getMap(getMapParamsDefaultResponseId, ApiType.PROCESSING);
+
+  expect(mockNetwork.history.post.length).toBe(1);
```

Comment:
This is only checking that the request was made - a much better test would be if you could check that the parameter was set correctly, because it would guard against regressions.

Something like that:
```javascript
    const request = mockNetwork.history.post[0];
    const requestData = JSON.parse(request.data);
    expect(requestData.output.responses).toEqual({ identifier: "default", format: "image/jpeg" });
```
(I didn't test it)

I think this should be fixed (and below), but no need to assign back when it's done, just resolve and merge.

Reply:
Thanks, I added tests like this. Will resolve and merge